include/sir/helpers.h

```diff
@@ -328,7 +328,7 @@ uint32_t FNV32_1a(const uint8_t* data, size_t len) {
     uint32_t hash = 2166136261U;
     for (size_t n = 0; n < len; n++) {
         hash ^= (uint32_t)data[n];
-        hash *= 16777619U;
+        hash = (uint32_t)(hash * 16777619ULL);
```

Comment:
I'm not sure I understand the utility in promoting the constant to 64-bit if we're to fit it in 32 bits. Shouldn't it be just `U`, or does adding ULL promote `hash` to unsigned long long as well?

Reply:
I'd have to check exactly what it was, but without it, it fails the sanitizer test due to an overflow.